lib/SGN/Controller/AJAX/BrAPI.pm

```diff
@@ -3319,17 +3319,31 @@ sub observationvariable_list_POST {
 	# TODO: auth later, not doing for now
 	# my ($auth,$user_id) = _authenticate_user($c);
 	my $user_id;
-	my $clean_inputs = $c->stash->{clean_inputs};
-	my $data = $clean_inputs;
-	my @all_variables;
-	foreach my $variable (values %{$data}) {
-		push @all_variables, $variable;
-	}
 
-	my $brapi = $self->brapi_module;
-	my $brapi_module = $brapi->brapi_wrapper('ObservationVariables');
-	my $brapi_package_result = $brapi_module->store(\@all_variables,$user_id, $c);
-	_standard_response_construction($c, $brapi_package_result);
+	my $response;
+	try {
+		my $clean_inputs = $c->stash->{clean_inputs};
+		my $data = $clean_inputs;
+		my @all_variables;
+		foreach my $variable (values %{$data}) {
+			push @all_variables, $variable;
+		}
+
+		my $brapi = $self->brapi_module;
+		my $brapi_module = $brapi->brapi_wrapper('ObservationVariables');
+		$response = $brapi_module->store(\@all_variables,$user_id, $c);
+	} catch {
+		if ($_->isa('CXGN::BrAPI::Exceptions::ConflictException')){
+			my $error = CXGN::BrAPI::JSONResponse->return_error([], $_->message);
+			_standard_response_construction($c, $error, 409);
+		} else {
+			warn Dumper($_);
```

Comment:
Yeah I wanted to keep that there so that we can see where the error came from. The Perl exceptions don't carry a stack trace like I was used to, so without the `warn` its hard to debug. It would be nice to clean this up in the future if we get the time.

Reply:
👍